src/components/ui/CustomModal.tsx

```diff
@@ -74,7 +82,11 @@ const CustomModal: FC<Props> = ({
               Icon={<CloseOutlined />}
             />
           </Box>
-          {Boolean(modalBody) && modalBody}
+          {modalBody && 
+           React.cloneElement(modalBody, {
+            onClose: handleClose,
```

Comment:
Why modalBody needs onClose?

Reply:
On form submission we need to close the modal thats why, onclose needed in the modalBody